Testing

rawatamit opened the pull request:
changes for dir structure, pom.xml, runs on IntelliJ without any issues, project structure should be correctly set.

bkushigian committed:
Update gitignore

Commit:
remove IDE settings

Commit:
Merge branch 'bug-fix' of https://github.com/bkushigian/controlflow

bkushigian committed:
Built basic tests

bkushigian committed:
Updated resources

bkushigian committed:
Bug fix

bkushigian committed:
Refactor TestUtil, fixed Test

    Cleaned up imports, fixed test value, TestingUtil refactor completion

Commit:
add more test case, need better test coverage

Commit:
more tests

Commit:
preserve mvn directory structure, fix pom.xml

- correct path for test dir as dictated by maven
- fixed pom.xml to remove redundant packages, and set UTF-8 as encoding for the sources
- revert controlflow code as original

bkushigian commented on src/test/java/de/codesourcery/asm/controlflow/ControlFlowAnalyzerTest.java:
```diff
@@ -0,0 +1,157 @@
+
+                    IBlock firstBlock = getFirstBlock(graph);
+                    IBlock firstBlockTrueOut = getTrueOutgoing(firstBlock);
+                    IBlock firstBlockFalseOut = getTrueOutgoing(firstBlock);
```
Typo? Should be `getFalseOutgoing()`?

bkushigian reviewed (changes requested):
Aside from the single typo noted below this looks good.

Commit:
add more test case for complex if statements
fix bug that was pointed out in the commit

Commit:
add class to simplify building CFG in code
port tests to use this class

Commit:
add comments

rawatamit commented:
Made changes, and changed a lot. Notes in commit.

bkushigian committed:
Failing test for isIso

Commit:
add new isomorphism test for CFGs
this commit is *not* intended to be merged
- new tests added
- new algorithm for isomorphism detection, proof in email
- DEBUG code still in the class
